pkg/statsd/parser.go

```diff
@@ -76,15 +76,20 @@ func (dp *DatagramParser) Run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case dgs := <-dp.in:
-			accumM, accumE, accumB := uint64(0), uint64(0), uint64(0)
+			var metrics []*gostatsd.Metric
+
+			accumB, accumE := uint64(0), uint64(0)
 			for _, dg := range dgs {
-				m, e, b := dp.handleDatagram(ctx, dg.Timestamp, dg.IP, dg.Msg)
+				parsedMetrics, e, b := dp.handleDatagram(ctx, dg.Timestamp, dg.IP, dg.Msg)
```

Comment:
I'm not loving this, because we're dispatching metrics externally, and events internally.  I'll probably clean that up later.